datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -93,6 +94,27 @@ pub trait PruningStatistics {
     ///
     /// Note: the returned array must contain [`Self::num_containers`] rows
     fn null_counts(&self, column: &Column) -> Option<ArrayRef>;
+
+    /// Returns an array where each row represents if the value of the
+    /// column is known to contain or not contain any of the set of `values`.
+    ///
+    /// This is used to prune containers using structures such as Bloom
+    /// Filters which can quickly test set membership.
+    ///
+    /// The returned array has one row for each container, with the following:
+    /// * `true` if the value of column CERTAINLY IS one of `values`
+    /// * `false` if the value of column CERTAINLY IS NOT one of  `values`
+    /// * `null` if the value of column may or may not be in values
+    ///
+    /// If these statistics can not determine column membership for any
+    /// container, return `None` (the default).
+    ///
+    /// Note: the returned array must contain [`Self::num_containers`] rows
+    fn contains(
```

Comment:
This is the new API -- it is slightly different than the proposal because it takes a `HashSet` rather than a single value, which is necessary to support `x IN (....)` type predicates